sklearn/covariance/_elliptic_envelope.py

```diff
@@ -163,6 +163,7 @@ def decision_function(self, X):
             compatibility with other outlier detection algorithms.
         """
         check_is_fitted(self)
+        X = self._validate_data(X, reset=False)
```

Comment:
Shall make the validation in `score_samples` instead since it is call below?
Right now, we don't have any check in this method.

Reply:
Yes. In this specific case and since `predict`, `decision_function`, `score` all call `score_samples`, let's move the validation to `score_samples`.